x-pack/plugins/rule_registry/server/utils/create_get_summarized_alerts_fn.ts

```diff
@@ -179,12 +180,41 @@ const getLifecycleAlertsByExecutionUuid = async ({
   };
 };
 
+const expandDottedField = (dottedFieldName: string, val: unknown): object => {
+  const parts = dottedFieldName.split('.');
+  if (parts.length === 1) {
+    return { [parts[0]]: val };
+  } else {
+    return { [parts[0]]: expandDottedField(parts.slice(1).join('.'), val) };
+  }
+};
+
+const expandFlattenedAlert = (alert: object) => {
+  if (Array.isArray(alert)) {
+    return alert;
```

Comment:
Could an alert-as-data record be an array?
I though it's always like:
```
{
'kibana.alert.id':'5'
'kibana.alert.name':'name'
}
```

Reply:
Probably not. I took this from detection rules library functions 🙈. I can remove the check

Reply:
I was going to check how security solutions did it :)